cylc/flow/task_events_mgr.py

```diff
@@ -1184,6 +1189,15 @@ def _process_job_logs_retrieval(
         # get a host to run retrieval on
         try:
             platform = get_platform(ctx.platform_name)
+        except PlatformLookupError:
+            log_platform_event(
+                'Unable to retrieve job logs.',
+                {'name': ctx.platform_name},
+                level='warning',
+            )
+            return
+
+        try:
             host = get_host_from_platform(platform, bad_hosts=self.bad_hosts)
```

Comment:
Split this `try` block into two so tools like mypy can tell that `platform` cannot be undefined.